mqt/qmap/subarchitectures.py

```diff
@@ -0,0 +1,275 @@
+        po_ref = defaultdict(lambda: [])
+        for k, v in po.items():
+            v_copy = v.copy()
+            v_copy.append(k)
+            po_ref[k] = v_copy
+        return po_ref
+
+    def __inverse_relation(self, po: PartialOrder) -> PartialOrder:
+        po_inv = defaultdict(lambda: [])
+        for k, v in po.items():
+            for e in v:
+                po_inv[e].append(k)
+        return po_inv
+
+    def __path_order_less(self, n, i, n_prime, i_prime) -> bool:
+        lhs = self.sgs[n][i]
+        rhs = self.sgs[n_prime][i_prime]
+        iso = self.__isomorphisms[(n, i)][(n_prime, i_prime)]
+        for v in range(lhs.num_nodes()):
+            for w in range(lhs.num_nodes()):
+                if v is w:
+                    continue
+                if (
+                    rx.dijkstra_shortest_path_lengths(lhs, v, lambda x: 1, goal=w)[w]
+                    > rx.dijkstra_shortest_path_lengths(rhs, iso[v], lambda x: 1, goal=iso[w])[iso[w]]
+                ):
+                    return True
+        return False
+
+    def __compute_desirable_subarchitectures(self):
+        self.__complete_isos()
+        for n in reversed(range(1, len(self.sgs[:-1]))):
+            for i in range(len(self.sgs[n])):
+                val = self.__isomorphisms[(n, i)]
+                # for (n, i), val in self.__isomorphisms.items():
+                for n_prime, i_prime in val.keys():
+                    if self.__path_order_less(n, i, n_prime, i_prime):
+                        self.desirable_subarchitectures[(n, i)].append((n_prime, i_prime))
+                des = self.desirable_subarchitectures[(n, i)].copy()
+                des.sort()
+                new_des = []
+                for j, (n_prime, i_prime) in enumerate(reversed(des)):
+                    j = len(des) - j - 1
+                    if not any(
+                        map(
+                            lambda k, n_prime=n_prime, i_prime=i_prime: (n_prime, i_prime) in self.subarch_order[k],
+                            des[:j],
+                        )
+                    ):
+                        new_des.append((n_prime, i_prime))
+
+                self.desirable_subarchitectures[(n, i)] = new_des
+                if len(self.desirable_subarchitectures[(n, i)]) == 0:
+                    self.desirable_subarchitectures[(n, i)].append((n, i))
+        self.desirable_subarchitectures[self.arch.num_nodes(), 0] = [(self.arch.num_nodes(), 0)]
+
+    def __cand(self, nqubits: int) -> set[Subarchitecture]:
+        all_desirables = [desirables for (n, i), desirables in self.desirable_subarchitectures.items() if n == nqubits]
+        return {des for desirables in all_desirables for des in desirables}
+
+
+def ibm_guadalupe_subarchitectures() -> SubarchitectureOrder:
+    """Load the precomputed ibm guadalupe subarchitectures."""
+    path = os.path.join(package_directory, "libs/ibm_guadalupe_16")
```

Comment:
See at the very top for the comment on using `importlib.resources`

Reply:
Done.